### PR TITLE
Handle CDATA in XML

### PR DESCRIPTION
**Pull request by camertron**
[PLAT-1357](https://lumoslabs.atlassian.net/browse/PLAT-1357)

Fixes the strange `<#cdata-section>` tags that are produced currently.

### CODE REVIEW

**Comment by elliottetzkorn** [comment]
If the conditional works for the example you have at the bottom, then 👍

File: spec/extractors/xml/fixtures.yml [hunk 1]
```diff
@@ -52,3 +52,6 @@
   entities.xml:
     - key: states
       string: "Alaska & Washington"
+  cdata.xml:
+    - key: foobar
+      string: "Hello <![CDATA[<b>]]>beautiful<![CDATA[</b>]]> world!"
```
Comment (elliottetzkorn) [inline spec/extractors/xml/fixtures.yml]: Actually would you change this string to: `"<![CDATA[Hello <b>beautiful</b> world!]]> "`
We're going to wrap the whole string in the cdata to make it more readable, and we only need the tag once as well.

File: spec/extractors/xml/fixtures/cdata.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<resources>
+    <string name="foobar">Hello <![CDATA[<b>]]>beautiful<![CDATA[</b>]]> world!</string>
```
Comment (elliottetzkorn) [inline spec/extractors/xml/fixtures/cdata.xml]: Same with this

**Review by ghost — CHANGES_REQUESTED** [review]

File: spec/extractors/xml/fixtures.yml (hunk 1 above)
Comment (camertron) [inline spec/extractors/xml/fixtures.yml]: I can add another test case.

**Comment by elliottetzkorn** [comment]
lgtm 🚢

File: lib/abroad/extractors/xml/xml_extractor.rb [hunk 3]
```diff
@@ -16,10 +16,8 @@ def extract_each(options = {}, &block)
         private
 
         def parse
-          Nokogiri::XML(stream) do |config|
-            # don't allow network connections
-            config.options = Nokogiri::XML::ParseOptions::NONET
-          end
+          options = Nokogiri::XML::ParseOptions.new.default_xml.nonet.noent
```
Comment (elliottetzkorn) [inline lib/abroad/extractors/xml/xml_extractor.rb]: actually i'm curious. what on earth does `nonet` followed by `noent` do?

Comment (camertron) [inline lib/abroad/extractors/xml/xml_extractor.rb]: Hehehehe good question. `nonet` disables network calls when parsing XML (the XML standard apparently provides for fetching remote XML resources during parse, who knew). `noent` means don't mess with HTML entities, although it doesn't appear to have any affect. I'll remove it.

Comment (elliottetzkorn) [inline lib/abroad/extractors/xml/xml_extractor.rb]: 👌

**Pull request merged**